src/components/Facility/ConsultationDoctorNotes/index.tsx

```diff
@@ -54,40 +55,52 @@ const ConsultationDoctorNotes = (props: ConsultationDoctorNotesProps) => {
 
   const initialData: PatientNoteStateType = {
     notes: [],
-    cPage: 1,
-    totalPages: 1,
     facilityId: facilityId,
     patientId: patientId,
   };
   const [state, setState] = useState(initialData);
+  const queryClient = useQueryClient();
+
+  const addNoteMutation = useMutation({
+    mutationFn: async (noteField: string) => {
+      const { res, data } = await request(routes.addPatientNote, {
+        pathParams: { patientId },
+        body: {
+          note: noteField,
+          thread,
+          consultation: consultationId,
+          reply_to: reply_to?.id,
+        },
+      });
+      if (res?.status === 201 && data) return data;
+      throw new Error("Failed to add note");
+    },
+    onSuccess: (newNote) => {
+      setState((prevState) => ({
+        ...prevState,
+        notes: [newNote, ...prevState.notes],
+      }));
+      setReplyTo(undefined);
+      Notification.Success({ msg: "Note added successfully" });
+
+      queryClient.invalidateQueries({
+        queryKey: ["notes"],
+      });
+    },
+    onError: () => {
+      Notification.Error({ msg: "Failed to add note" });
+    },
+  });
 
-  const onAddNote = async () => {
+  const onAddNote = () => {
     if (!/\S+/.test(noteField)) {
       Notification.Error({
         msg: "Note Should Contain At Least 1 Character",
       });
       return;
     }
-
-    const { res } = await request(routes.addPatientNote, {
-      pathParams: {
-        patientId: patientId,
-      },
-      body: {
-        note: noteField,
-        thread,
-        consultation: consultationId,
-        reply_to: reply_to?.id,
-      },
-    });
-
-    if (res?.status === 201) {
-      Notification.Success({ msg: "Note added successfully" });
-      setState({ ...state, cPage: 1 });
-      setNoteField("");
-      setReload(true);
-      setReplyTo(undefined);
-    }
+    addNoteMutation.mutate(noteField);
+    setNoteField("");
```

Comment:
pass this as `onSuccess` arg. for the useMutation instead.

Reply:
Just setNoteField right?